sklearn/gaussian_process/_gpr.py

```diff
@@ -358,18 +355,12 @@ def predict(self, X, return_std=False, return_cov=False):
 
                 return y_mean, y_cov
             elif return_std:
-                # cache result of K_inv computation
-                if self._K_inv is None:
-                    # compute inverse K_inv of K based on its Cholesky
-                    # decomposition L and its inverse L_inv
-                    L_inv = solve_triangular(self.L_.T,
-                                             np.eye(self.L_.shape[0]))
-                    self._K_inv = L_inv.dot(L_inv.T)
+                v = cho_solve((self.L_, True), K_trans.T)  # Line 5
 
                 # Compute variance of predictive distribution
+                # Use einsum to avoid large matrix
```

Comment:
```suggestion
                # Use einsum to avoid explicitly forming the large matrix
                # K_trans @ v just to extract its diagonal afterward.
```